application-mocca-calendar-ui/src/main/resources/MoccaCalendar/Code/UpcomingEventsMacro.xml

```diff
@@ -0,0 +1,623 @@
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.StringClass</classType>
+      </id>
+      <name>
+        <disabled>0</disabled>
+        <name>name</name>
+        <number>2</number>
+        <prettyName>Macro name</prettyName>
+        <size>30</size>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.StringClass</classType>
+      </name>
+      <supportsInlineMode>
+        <disabled>0</disabled>
+        <displayFormType>select</displayFormType>
+        <displayType>yesno</displayType>
+        <name>supportsInlineMode</name>
+        <number>5</number>
+        <prettyName>Supports inline mode</prettyName>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.BooleanClass</classType>
+      </supportsInlineMode>
+      <visibility>
+        <cache>0</cache>
+        <disabled>0</disabled>
+        <displayType>select</displayType>
+        <freeText>forbidden</freeText>
+        <largeStorage>0</largeStorage>
+        <multiSelect>0</multiSelect>
+        <name>visibility</name>
+        <number>6</number>
+        <prettyName>Macro visibility</prettyName>
+        <relationalStorage>0</relationalStorage>
+        <separator>|</separator>
+        <separators>|</separators>
+        <size>1</size>
+        <unmodifiable>0</unmodifiable>
+        <values>Current User|Current Wiki|Global</values>
+        <classType>com.xpn.xwiki.objects.classes.StaticListClass</classType>
+      </visibility>
+    </class>
+    <property>
+      <async_cached>0</async_cached>
+    </property>
+    <property>
+      <async_context/>
+    </property>
+    <property>
+      <async_enabled>0</async_enabled>
+    </property>
+    <property>
+      <code>{{velocity}}
+#set ($discard = $xwiki.ssx.use('MoccaCalendar.Code.UpcomingEventsMacro'))
+#set ($startDate = $xwiki.jodatime.getDateTime())
+#set ($endDate = $startDate.plusYears(1))
+#set ($dateTimeFormatter = $xwiki.jodatime.getDateTimeFormatterForPattern('dd/MM/yyyy HH:mm'))
+#set ($dateFormatter = $xwiki.jodatime.getDateTimeFormatterForPattern('dd/MM/yyyy'))
+#set ($timeFormatter = $xwiki.jodatime.getDateTimeFormatterForPattern('HH:mm'))
+#set ($dayFormatter = $xwiki.jodatime.getDateTimeFormatterForPattern('d').withLocale($xcontext.language))
+#set ($monthFormatter = $xwiki.jodatime.getDateTimeFormatterForPattern('MMM').withLocale($xcontext.language))
+#set ($wikiList = [])
+#set ($wikis = "$!wikimacro.parameters.wikis")
+#if ($wikis != '')
+  #foreach ($wikiId in $wikis.split(','))
+    #set ($discard = $wikiList.add($wikiId.trim()))
+  #end
+#else
+  #set ($discard = $wikiList.add($xcontext.wiki))
+#end
+#set ($events = $services.moccacalendar.queryEvents($startDate.toDate(), $endDate.toDate(), $wikiList, true))
+(% class="moccacal-event-list" %)(((
+  #foreach ($event in $events)
+    #if ($wikimacro.parameters.limit &gt; 0 &amp;&amp; $foreach.count &gt; $wikimacro.parameters.limit)
```

Comment:
I suppose ``queryEvents`` doesn't support limiting the results? I find this strange.

Reply:
That would be a useful improvement indeed, I created #93 about it.